hazelcast/src/main/java/com/hazelcast/map/impl/MapStoreWrapper.java

```diff
@@ -174,11 +173,26 @@ public Map loadAll(Collection keys) {
             return Collections.EMPTY_MAP;
         }
         if (isMapLoader()) {
-            return mapLoader.loadAll(keys);
+            Map mapLoaded = mapLoader.loadAll(keys);
+            validateLoadedMap(mapLoaded, keys);
+            return mapLoaded;
         }
         return null;
     }
 
+    private void validateLoadedMap(Map mapLoaded, Collection expectedKeys) {
+        if (mapLoaded != null) {
+            for (Object loadedKey : mapLoaded.keySet()) {
+                if (!expectedKeys.contains(loadedKey)) {
```

Comment:
Seems it is legal to expect some of the keys cannot be loaded(because they don't exist in database at the time of loading). Maybe as an alternative solution we can check partitions of the keys returned from `MapDataStore#loadAll` for the fix?